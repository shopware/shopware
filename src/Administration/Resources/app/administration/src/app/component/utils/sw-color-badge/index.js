import template from './sw-color-badge.html.twig';
import './sw-color-badge.scss';

const { Component } = Shopware;

/**
 * @public
 * @description
 * Renders a colored badge for example as indicator if an item is available.
 * @status ready
 * @example-type static
 * @component-example
 * <div>
 *     <!-- red square -->
 *     <sw-color-badge color="red"></sw-color-badge>
 *     <!-- green square -->
 *     <sw-color-badge color="green"></sw-color-badge>
 *     <!-- red circle -->
 *     <sw-color-badge color="red" rounded></sw-color-badge>
 * </div>
 */
Component.register('sw-color-badge', {
    template,

    props: {
        variant: {
            type: String,
            required: false,
            default: 'default'
        },
        color: {
            type: String,
            required: false,
            default: ''
        },
        rounded: {
            type: Boolean,
            required: false,
            default: false
        }
    },

    computed: {
        colorStyle() {
            if (!this.color.length) {
                return '';
            }
            return `background:${this.color}`;
        },
        variantClass() {
            return {
                [`is--${this.variant}`]: true,
                'is--rounded': this.rounded
            };
        }
    }
});
