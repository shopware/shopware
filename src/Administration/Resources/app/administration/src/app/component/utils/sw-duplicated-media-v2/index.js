import template from './sw-duplicated-media-v2.html.twig';
import './sw-duplicated-media-v2.scss';

const { Component, Context, Filter } = Shopware;
const { Criteria } = Shopware.Data;

/**
 * @private
 */

const LOCAL_STORAGE_KEY_OPTION = 'sw-duplicate-media-resolve-option';
const LOCAL_STORAGE_SAVE_SELECTION = 'sw-duplicate-media-resolve-save-selection';

Component.register('sw-duplicated-media-v2', {
    template,

    inject: ['repositoryFactory', 'mediaService'],

    data() {
        return {
            isLoading: false,
            shouldSaveSelection: false,
            selectedOption: 'Replace',
            suggestedName: '',
            existingMedia: null,
            targetEntity: null,
            failedUploadTasks: [],
            postponedFailedUploads: [],
        };
    },

    computed: {
        mediaRepository() {
            return this.repositoryFactory.create('media');
        },

        additionalErrorCount() {
            return this.failedUploadTasks.length - 1;
        },

        hasAdditionalErrors() {
            return this.additionalErrorCount > 0;
        },

        currentTask() {
            return this.failedUploadTasks[0];
        },

        buttonLabel() {
            return this.$tc(`global.sw-duplicated-media-v2.button${this.selectedOption}`);
        },

        dateFilter() {
            return Filter.getByName('date');
        },

        fileSizeFilter() {
            return Filter.getByName('fileSize');
        },

        currentTaskDetails() {
            if (!this.currentTask) {
                return '';
            }
            const metadata = [
                this.dateFilter(new Date(), { month: 'long' }),
            ];

            if (this.currentTask.src instanceof File) {
                metadata.push(this.fileSizeFilter(this.currentTask.src.size));
            }

            return metadata.join(', ');
        },

        showModal() {
            return this.failedUploadTasks.length > 0 && !this.isWorkingOnMultipleTasks;
        },

        isWorkingOnMultipleTasks() {
            return this.isLoading && this.shouldSaveSelection;
        },

        options() {
            return [
                {
                    value: 'Replace',
                    name: this.$tc('global.sw-duplicated-media-v2.labelOptionReplace'),
                },
                {
                    value: 'Rename',
                    name: this.$tc('global.sw-duplicated-media-v2.labelOptionRename'),
                },
                {
                    value: 'Skip',
                    name: this.$tc('global.sw-duplicated-media-v2.labelOptionSkip'),
                },
            ];
        },
    },

    watch: {
        currentTask() {
            this.updatePreviewData();
        },

        showModal(newVal) {
            if (newVal) {
                this.loadDefaultOption();
                return;
            }

            this.saveDefaultOption();
        },

        isLoading(newVal) {
            if (newVal) {
                return;
            }

            this.failedUploadTasks.push(...this.postponedFailedUploads.splice(0, this.postponedFailedUploads.length));
        },
    },

    created() {
        this.createdComponent();
    },

    beforeDestroy() {
        this.beforeDestroyComponent();
    },

    methods: {
        createdComponent() {
            this.loadDefaultOption();
            this.updatePreviewData();

            this.mediaService.addDefaultListener(this.handleMediaServiceUploadEvent);
        },

        beforeDestroyComponent() {
            this.mediaService.removeDefaultListener(this.handleMediaServiceUploadEvent);
        },

        loadDefaultOption() {
            this.shouldSaveSelection = localStorage.getItem(LOCAL_STORAGE_SAVE_SELECTION) || false;

            if (this.shouldSaveSelection) {
                this.defaultOption = localStorage.getItem(LOCAL_STORAGE_KEY_OPTION) || 'Replace';
            }

            this.selectedOption = this.defaultOption || 'Replace';
        },

        saveDefaultOption() {
            localStorage.setItem(LOCAL_STORAGE_SAVE_SELECTION, this.shouldSaveSelection);

            if (this.shouldSaveSelection) {
                localStorage.setItem(LOCAL_STORAGE_KEY_OPTION, this.defaultOption);
            }
        },

        handleMediaServiceUploadEvent({ action, payload }) {
            if (action !== 'media-upload-fail') {
                return;
            }

            if (!this.isDuplicatedNameError(payload.error)) {
                return;
            }

            if (this.isLoading) {
                this.postponedFailedUploads.push(payload);
                return;
            }

            this.failedUploadTasks.push(payload);
        },

        isDuplicatedNameError(error) {
            return error.response.data.errors.some((err) => {
                return err.code === 'CONTENT__MEDIA_DUPLICATED_FILE_NAME';
            });
        },

        async updatePreviewData() {
            if (!this.currentTask) {
                this.existingMedia = null;
                this.suggestedName = '';
                return;
            }

            const criteria = new Criteria(1, 1)
                .addFilter(
                    Criteria.multi(
                        'AND',
                        [
                            Criteria.equals('fileName', this.currentTask.fileName),
                            Criteria.equals('fileExtension', this.currentTask.extension),
                        ],
                    ),
                );

            const searchResult = await this.mediaRepository.search(criteria, Context.api);
            if (searchResult?.[0]) {
                this.existingMedia = searchResult[0];
            }

            const provided = await this.mediaService.provideName(this.currentTask.fileName, this.currentTask.extension);
            this.suggestedName = provided.fileName;
        },

        solveDuplicate() {
            if (!this.currentTask) {
                this.isLoading = false;
                return;
            }

            this.isLoading = true;

            switch (this.selectedOption) {
                case 'Rename':
                    this.renameFile(this.currentTask);
                    break;
                case 'Replace':
                    this.replaceFile(this.currentTask);
                    break;
                case 'Skip':
                default:
                    this.skipFile(this.currentTask);
                    break;
            }

            this.failedUploadTasks.splice(0, 1);

            if (!this.currentTask) {
                this.isLoading = false;
            } else {
                this.solveDuplicate();
            }
        },

        async renameFile(uploadTask) {
            const newTask = Object.assign({}, uploadTask);

            const { fileName } = await this.mediaService.provideName(uploadTask.fileName, uploadTask.extension);
            newTask.fileName = fileName;

            this.mediaService.addUpload(newTask.uploadTag, newTask);
            await this.mediaService.runUploads(newTask.uploadTag);
        },

        skipAll() {
            this.isLoading = true;

            this.skipFile(this.currentTask);
            this.failedUploadTasks.splice(0, 1);

            if (!this.currentTask) {
                this.isLoading = false;
            } else {
                this.skipAll();
            }
        },

        skipCurrentFile() {
            this.isLoading = true;
            this.skipFile(this.currentTask);

            this.failedUploadTasks.splice(0, 1);
            this.isLoading = false;
        },

        async skipFile(uploadTask) {
            const oldTarget = await this.mediaRepository.get(uploadTask.targetId, Context.api);
            if (!oldTarget.hasFile) {
                await this.mediaRepository.delete(oldTarget.id, Context.api);
            }

            this.mediaService.cancelUpload(uploadTask.uploadTag, uploadTask);
        },

        async replaceFile(uploadTask) {
            const criteria = new Criteria(1, 1)
                .addFilter(Criteria.multi('AND',
                    [
                        Criteria.equals('fileName', uploadTask.fileName),
                        Criteria.equals('fileExtension', uploadTask.extension),
                    ]));

            const searchResult = await this.mediaRepository.search(criteria, Context.api);
            const newTarget = searchResult[0];
            const oldTargetId = uploadTask.targetId;
            uploadTask.targetId = newTarget.id;

            this.mediaService.addUpload(uploadTask.uploadTag, uploadTask);

            await this.mediaService.runUploads(uploadTask.uploadTag);
            const oldTarget = await this.mediaRepository.get(oldTargetId, Context.api);

            if (!oldTarget.hasFile) {
                await this.mediaRepository.delete(oldTargetId, Context.api);
            }

            await this.mediaRepository.get(uploadTask.targetId, Context.api);
        },
    },
});
