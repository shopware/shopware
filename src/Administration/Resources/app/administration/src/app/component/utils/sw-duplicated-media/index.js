import CriteriaFactory from 'src/core/factory/criteria.factory';
import template from './sw-duplicated-media.html.twig';
import './sw-duplicated-media.scss';

const { Component, StateDeprecated, Filter } = Shopware;

/**
 * @private
 */

const LOCAL_STORAGE_KEY_OPTION = 'sw-duplicate-media-resolve-option';
const LOCAL_STORAGE_SAVE_SELECTION = 'sw-duplicate-media-resolve-save-selection';

Component.register('sw-duplicated-media', {
    template,

    inject: [
        'mediaService'
    ],

    data() {
        return {
            isLoading: false,
            shouldSaveSelection: false,
            selectedOption: 'Replace',
            suggestedName: '',
            existingMedia: null,
            targetEntity: null,
            failedUploadTasks: [],
            postponedFailedUploads: []
        };
    },

    computed: {
        mediaStore() {
            return StateDeprecated.getStore('media');
        },

        uploadStore() {
            return StateDeprecated.getStore('upload');
        },

        additionalErrorCount() {
            return this.failedUploadTasks.length - 1;
        },

        hasAdditionalErrors() {
            return this.additionalErrorCount > 0;
        },

        currentTask() {
            return this.failedUploadTasks[0];
        },

        buttonLabel() {
            return this.$tc(`global.sw-duplicated-media.button${this.selectedOption}`);
        },

        dateFilter() {
            return Filter.getByName('date');
        },

        fileSizeFilter() {
            return Filter.getByName('fileSize');
        },

        currentTaskDetails() {
            if (!this.currentTask) {
                return '';
            }
            const metadata = [
                this.dateFilter(new Date(), { month: 'long' })
            ];

            if (this.currentTask.src instanceof File) {
                metadata.push(this.fileSizeFilter(this.currentTask.src.size));
            }

            return metadata.join(', ');
        },

        showModal() {
            return this.failedUploadTasks.length > 0 && !this.isWorkingOnMultipleTasks;
        },

        isWorkingOnMultipleTasks() {
            return this.isLoading && this.shouldSaveSelection;
        },

        options() {
            return [
                {
                    value: 'Replace',
                    name: this.$tc('global.sw-duplicated-media.labelOptionReplace')
                },
                {
                    value: 'Rename',
                    name: this.$tc('global.sw-duplicated-media.labelOptionRename')
                },
                {
                    value: 'Skip',
                    name: this.$tc('global.sw-duplicated-media.labelOptionSkip')
                }
            ];
        }
    },

    watch: {
        currentTask() {
            this.updatePreviewData();
        },

        showModal(newVal) {
            if (newVal) {
                this.loadDefaultOption();
                return;
            }

            this.saveDefaultOption();
        },

        isLoading(newVal) {
            if (newVal) {
                return;
            }

            this.failedUploadTasks.push(...this.postponedFailedUploads.splice(0, this.postponedFailedUploads.length));
        }
    },

    created() {
        this.createdComponent();
    },

    beforeDestroy() {
        this.beforeDestroyComponent();
    },

    methods: {
        createdComponent() {
            this.loadDefaultOption();
            this.updatePreviewData();

            this.uploadStore.addDefaultListener(this.handleUploadStoreEvent);
        },

        beforeDestroyComponent() {
            this.uploadStore.removeDefaultListener(this.handleUploadStoreEvent);
        },

        loadDefaultOption() {
            this.shouldSaveSelection = localStorage.getItem(LOCAL_STORAGE_SAVE_SELECTION) || false;
            if (this.shouldSaveSelection) {
                this.defaultOption = localStorage.getItem(LOCAL_STORAGE_KEY_OPTION) || 'Replace';
            }
            this.selectedOption = this.defaultOption || 'Replace';
        },

        saveDefaultOption() {
            localStorage.setItem(LOCAL_STORAGE_SAVE_SELECTION, this.shouldSaveSelection);
            if (this.shouldSaveSelection) {
                localStorage.setItem(LOCAL_STORAGE_KEY_OPTION, this.defaultOption);
            }
        },

        handleUploadStoreEvent({ action, payload }) {
            if (action !== 'media-upload-fail') {
                return;
            }

            if (!this.isDuplicatedNameError(payload.error)) {
                return;
            }

            if (this.isLoading) {
                this.postponedFailedUploads.push(payload);
                return;
            }

            this.failedUploadTasks.push(payload);
        },

        isDuplicatedNameError(error) {
            return error.response.data.errors.some((err) => {
                return err.code === 'CONTENT__MEDIA_DUPLICATED_FILE_NAME';
            });
        },

        updatePreviewData() {
            if (!this.currentTask) {
                this.existingMedia = null;
                this.suggestedName = '';
                return;
            }

            this.mediaStore.getList({
                page: 1,
                limit: 1,
                criteria: CriteriaFactory.multi('AND',
                    CriteriaFactory.equals('fileName', this.currentTask.fileName),
                    CriteriaFactory.equals('fileExtension', this.currentTask.extension))
            }).then((response) => {
                this.existingMedia = response.items[0];
            });

            this.mediaService.provideName(this.currentTask.fileName, this.currentTask.extension).then((response) => {
                this.suggestedName = response.fileName;
            });
        },

        solveDuplicate() {
            if (!this.currentTask) {
                this.isLoading = false;
                return Promise.resolve();
            }

            this.isLoading = true;
            let solvingPromise = null;
            switch (this.selectedOption) {
                case 'Rename':
                    solvingPromise = this.renameFile(this.currentTask);
                    break;
                case 'Replace':
                    solvingPromise = this.replaceFile(this.currentTask);
                    break;
                case 'Skip':
                default:
                    solvingPromise = this.skipFile(this.currentTask);
                    break;
            }

            return solvingPromise.then(() => {
                this.failedUploadTasks.splice(0, 1);
                if (this.shouldSaveSelection) {
                    return this.solveDuplicate();
                }
                this.isLoading = false;
                return Promise.resolve();
            }).catch((cause) => {
                this.isLoading = false;
                return Promise.reject(cause);
            });
        },

        renameFile(uploadTask) {
            const newTask = Object.assign({}, uploadTask);
            return this.mediaService.provideName(uploadTask.fileName, uploadTask.extension).then(({ fileName }) => {
                newTask.fileName = fileName;
                this.uploadStore.addUpload(newTask.uploadTag, newTask);
                return this.uploadStore.runUploads(newTask.uploadTag);
            });
        },

        skipAll() {
            this.isLoading = true;

            return this.skipFile(this.currentTask).then(() => {
                this.failedUploadTasks.splice(0, 1);
                if (this.currentTask) {
                    return this.skipAll();
                }

                this.isLoading = false;
                return Promise.resolve();
            }).catch(() => {
                this.isLoading = false;
                return Promise.reject();
            });
        },

        skipCurrentFile() {
            this.isLoading = true;
            this.skipFile(this.currentTask).then(() => {
                this.failedUploadTasks.splice(0, 1);
                this.isLoading = false;
            }).catch(() => {
                this.isLoading = false;
            });
        },

        skipFile(uploadTask) {
            return this.mediaStore.getByIdAsync(uploadTask.targetId).then((oldTarget) => {
                if (!oldTarget.hasFile) {
                    oldTarget.delete(true);
                }
            });
        },

        replaceFile(uploadTask) {
            return this.mediaStore.getList({
                page: 1,
                limit: 1,
                criteria: CriteriaFactory.multi('AND',
                    CriteriaFactory.equals('fileName', uploadTask.fileName),
                    CriteriaFactory.equals('fileExtension', uploadTask.extension))
            }).then((response) => {
                const newTarget = response.items[0];
                const oldTargetId = uploadTask.targetId;
                uploadTask.targetId = newTarget.id;

                this.uploadStore.addUpload(uploadTask.uploadTag, uploadTask);
                return this.uploadStore.runUploads(uploadTask.uploadTag).then(() => {
                    return this.mediaStore.getByIdAsync(oldTargetId).then((oldTarget) => {
                        if (!oldTarget.hasFile) {
                            oldTarget.delete(true);
                        }
                        return this.mediaStore.getByIdAsync(uploadTask.targetId);
                    });
                });
            });
        }
    }
});
