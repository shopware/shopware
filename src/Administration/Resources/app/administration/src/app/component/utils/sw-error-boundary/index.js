/**
 * @package admin
 *
 * @private
 * @description
 * The component catches all errors in subcomponent which aren't handled before.
 * @status ready
 * @example-type code-only
 * @component-example
 * <sw-error-boundary>
 *     <!-- Your components -->
 * </sw-error-boundary>
 */
Shopware.Component.register('sw-error-boundary', {

    render() {
        return this.$slots.default;
    },

    inject: ['repositoryFactory'],

    computed: {
        logEntryRepository() {
            return this.repositoryFactory.create('log_entry', null, {
                keepApiErrors: true,
            });
        },
    },

    errorCaptured(err, vm) {
        console.error('An error was captured in current module:', err);

        this.logErrorInEntries(err, vm);

        // stop error propagation
        return false;
    },

    methods: {
        logErrorInEntries(err, vm) {
            if (!err) {
                return;
            }

            const newLogEntry = this.logEntryRepository.create();

            newLogEntry.message = err.toString();
            newLogEntry.channel = 'Administration';
            newLogEntry.level = 400;
            newLogEntry.context = {
                component: vm?._name ?? 'Unknown component',
                stack: err.stack ?? 'Unknown stack',
                url: window.location.href,
            };

            this.logEntryRepository.save(newLogEntry).catch(e => Shopware.Utils.debug.error(e));
        },
    },
});
