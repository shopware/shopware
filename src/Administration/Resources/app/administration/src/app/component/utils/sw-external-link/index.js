import template from './sw-external-link.html.twig';
import './sw-external-link.scss';

const { Component } = Shopware;

/**
 * @package admin
 *
 * @deprecated tag:v6.6.0 - Will be private
 * @public
 * @description Link to another website outside the admin, that opens in a new browser tab
 * @status ready
 * @example-type dynamic
 * @component-example
 * <sw-external-link
 *   href="https://google.com">
 *   Ask google
 * </sw-external-link>
 */
// eslint-disable-next-line sw-deprecation-rules/private-feature-declarations
Component.register('sw-external-link', {
    template,

    inheritAttrs: false,

    props: {
        small: {
            type: Boolean,
            required: false,
            default: false,
        },

        icon: {
            type: String,
            required: false,
            default: 'regular-external-link-s',
        },

        rel: {
            type: String,
            required: false,
            default: 'noopener',
        },
    },

    computed: {
        classes() {
            return {
                'sw-external-link--small': this.small,
            };
        },

        iconSize() {
            if (this.small) {
                return '8px';
            }

            return '10px';
        },
    },

    methods: {
        onClick(event) {
            this.$emit('click', event);
        },
    },
});
