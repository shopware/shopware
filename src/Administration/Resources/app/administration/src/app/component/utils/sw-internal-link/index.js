import template from './sw-internal-link.html.twig';
import './sw-internal-link.scss';

const { Component } = Shopware;

/**
 * @public
 * @description Link to another route inside the administration
 * @status ready
 * @example-type dynamic
 * @component-example
 * <sw-internal-link
 *   :routerLink="{ name: "sw.product.index" }">
 *   Go to products
 * </sw-internal-link>
 */
Component.register('sw-internal-link', {
    template,

    props: {
        routerLink: {
            type: Object,
            required: true,
        },

        target: {
            type: String,
            required: false,
            default: null,
        },

        icon: {
            type: String,
            required: false,
            default: 'default-arrow-simple-right',
        },

        inline: {
            type: Boolean,
            required: false,
            default: false,
        },
    },

    computed: {
        componentClasses() {
            return {
                'sw-internal-link--inline': this.inline,
            };
        },
    },
});
