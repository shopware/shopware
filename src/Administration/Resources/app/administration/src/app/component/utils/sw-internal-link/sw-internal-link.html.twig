<!-- eslint-disable-next-line sw-deprecation-rules/no-twigjs-blocks -->
{% block sw_internal_link %}
<router-link
    v-if="!disabled"
    :to="routerLink"
    :target="target"
    class="sw-internal-link"
    :class="componentClasses"
>

    <!-- eslint-disable-next-line sw-deprecation-rules/no-twigjs-blocks -->
    {% block sw_internal_link__slot %}
    <slot></slot>
    {% endblock %}

    <!-- eslint-disable-next-line sw-deprecation-rules/no-twigjs-blocks -->
    {% block sw_internal_link__icon %}
    <sw-icon
        v-if="!hideIcon"
        :name="icon"
        small
    />
    {% endblock %}

</router-link>

<span
    v-else
    class="sw-internal-link sw-internal-link--disabled"
>

    <!-- eslint-disable-next-line sw-deprecation-rules/no-twigjs-blocks -->
    {% block sw_internal_link__slot %}
    <slot></slot>
    {% endblock %}

    <!-- eslint-disable-next-line sw-deprecation-rules/no-twigjs-blocks -->
    {% block sw_internal_link__icon %}
    <sw-icon
        v-if="!hideIcon"
        :name="icon"
        small
    />
    {% endblock %}

</span>
{% endblock %}
