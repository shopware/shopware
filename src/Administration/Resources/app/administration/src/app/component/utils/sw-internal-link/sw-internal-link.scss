@import "~scss/variables";

.sw-internal-link {
    display: grid;
    grid-auto-flow: column;
    grid-column-gap: 6px;
    align-items: center;
    text-decoration: none;
    color: $color-shopware-brand-500;
    font-size: $font-size-small;

    &:hover,
    &:focus {
        color: $color-shopware-brand-600;
    }

    &--inline {
        display: inline;
    }

    &--disabled {
        color: $color-gray-500;
    }
}
