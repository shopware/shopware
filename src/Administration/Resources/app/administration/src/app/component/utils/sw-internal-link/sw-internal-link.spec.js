/**
 * @package admin
 */

import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import 'src/app/component/utils/sw-internal-link';

// initial component setup
const setup = async (propOverride) => {
    const propsData = {
        routerLink: { name: 'sw.product.index' },
        ...propOverride
    };

    return shallowMount(await Shopware.Component.build('sw-internal-link'), {
        stubs: {
            'sw-icon': true,
            RouterLink: RouterLinkStub
        },
        slots: {
            default: 'test internal link'
        },
        propsData
    });
};

describe('components/utils/sw-internal-link', () => {
    it('should be a Vue.js component', async () => {
        const wrapper = await setup();
        expect(wrapper.vm).toBeTruthy();
    });

    it('should render correctly', async () => {
        const wrapper = await setup();
        expect(wrapper.element).toMatchSnapshot();
    });

    it('should render correctly when disabled', async () => {
        const wrapper = await setup({ disabled: true });
        expect(wrapper.element).toMatchSnapshot();
    });

    it('should display a custom icon', async () => {
        const wrapper = await setup({ icon: 'default-test-icon' });

        expect(wrapper.find('sw-icon-stub').attributes().name).toBe('default-test-icon');
    });

    it('should add custom target to link', async () => {
        const wrapper = await setup({ target: '_blank' });

        expect(wrapper.findComponent(RouterLinkStub).props().to).toEqual({ name: 'sw.product.index' });
    });

    it('should add inline class if it is an inline link', async () => {
        const wrapper = await setup({ inline: true });

        expect(wrapper.findComponent(RouterLinkStub).classes()).toContain('sw-internal-link--inline');
    });
});
