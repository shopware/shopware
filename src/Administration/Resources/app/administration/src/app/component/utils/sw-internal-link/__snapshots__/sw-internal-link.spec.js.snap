// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/utils/sw-internal-link should render correctly 1`] = `
<a
  class="sw-internal-link"
>
  test internal link 
  <sw-icon-stub
    name="regular-long-arrow-right"
    small=""
  />
</a>
`;

exports[`components/utils/sw-internal-link should render correctly when disabled 1`] = `
<span
  class="sw-internal-link sw-internal-link--disabled"
>
  test internal link 
  <sw-icon-stub
    name="regular-long-arrow-right"
    small=""
  />
</span>
`;
